microsoft-edge/progressive-web-apps-chromium/how-to/index.md

```diff
@@ -138,63 +164,208 @@ Your VS Code project should now look somewhat like this:
+
+    ```html
+    <script src="converter.js"></script>
+    ```
+
+1. Now add some CSS style to the app, to make it more visually interesting. Create a new file called `converter.css` in your project and add the following code to it:
 
-Service workers are defined in a special JavaScript file, described in [Using Service Workers](https://developer.mozilla.org/docs/Web/API/Service_Worker_API/Using_Service_Workers) and [Service Worker API](https://developer.mozilla.org/docs/Web/API/Service_Worker_API).
+    ```css
+    html {
+      background: rgb(243, 243, 243);
+      font-family: system-ui, -apple-system, BlinkMacSystemFont, 'Segoe UI', Roboto, Oxygen, Ubuntu, Cantarell, 'Open Sans', 'Helvetica Neue', sans-serif;
+      font-size: 15pt;
+    }
+    
+    html, body {
+      height: 100%;
+      margin: 0;
+    }
+    
+    body {
+      display: grid;
+      place-items: center;
+    }
+    
+    #converter {
+      width: 15rem;
+      padding: 2rem;
+      border-radius: .5rem;
+      box-shadow: 0 0 2rem 0 #0001;
+      display: flex;
+      flex-direction: column;
+      align-items: center;
+    }
+    
+    #converter input, #converter select {
+      font-family: inherit;
+      font-size: inherit;
+      margin-block-end: 1rem;
+      text-align: center;
+      width: 10rem;
+    }
+    
+    #converter #output-temp {
+      font-size: 2rem;
+      font-weight: bold;
+    }
+    ```
 
-To add a service worker in your project:
+1. Open `index.html` again and reference the new CSS file in it by adding the following code inside the `<head>` tag:
+
+    ```html
+    <link rel="stylesheet" href="converter.css">
+    ```
+
+Your VS Code project should now look something like this
+
+![Screenshot of VS Code showing the sample PWA project, with the index.html, converter.js, converter.css, and manifest.json files](../media/visual-studio-project-with-front-end-files.png)
+
+Now browse to `http://localhost:8080` to view your app.
+
+![Running your new PWA, with the frontend code, on localhost.](../media/sample-pwa-app-with-frontend-code.png)
+
+Your app does something useful now but can't be installed yet. You will make your app installable in the next step by creating a service worker.
+
+
+<!-- ====================================================================== -->
+## Step 5 - Add a service worker
+
+Service workers are a key technology that help make PWAs faster and independent of network conditions.
+
+Service workers are specialized [Web Workers](https://developer.mozilla.org/docs/Web/API/Web_Workers_API) that intercept network requests from your PWA and enable scenarios that were previously limited to native apps, including:
+
+*  Offline support.
+*  Advanced caching.
+*  Running background tasks such as receiving PUSH messages, adding badges to the app icon, or fetching data from a server.
+
+Your app must have a service worker file for Microsoft Edge to install the app.
```

Comment:
```suggestion
For Microsoft Edge to be able to install the app, your app must have a service worker file.
```
that's to prevent this mis-parsing:
Your app needs:
* A service worker file for Microsoft Edge.